src/libos/crates/vdso-time/src/lib.rs

```diff
@@ -9,289 +7,284 @@ extern crate sgx_types;
+        let vdso_addr = Self::get_checked_vdso_addr(vdso_addr)?;
 
         // release, e.g., "5.9.6-050906-generic"
-        let release = unsafe { &*(&release as *const [i8] as *const [u8]) };
+        let release = unsafe { &*(release as *const [i8] as *const [u8]) };
         let release = str::from_utf8(release);
         if release.is_err() {
-            return Err(());
+            return_errno!(EINVAL, "Vdso get kernel release failed")
         }
         let mut release = release.unwrap().split(&['-', '.', ' '][..]);
-        let version_big: u8 = release.next().unwrap_or("0").parse().unwrap_or(0);
-        let version_little: u8 = release.next().unwrap_or("0").parse().unwrap_or(0);
-
-        let vdso_data_ptr = match (version_big, version_little) {
+        let version_big: u8 = release
+            .next()
+            .ok_or(errno!(EINVAL, "Vdso get kernel big version failed"))?
+            .parse()?;
+        let version_little: u8 = release
+            .next()
+            .ok_or(errno!(EINVAL, "Vdso get kernel little version failed"))?
+            .parse()?;
+
+        Ok(match (version_big, version_little) {
             (4, 0..=4) | (4, 7..=11) => VdsoDataPtr::V4_0(vdso_data_v4_0::vdsodata_ptr(vdso_addr)),
             (4, 5..=6) | (4, 12..=19) => VdsoDataPtr::V4_5(vdso_data_v4_5::vdsodata_ptr(vdso_addr)),
             (5, 0..=2) => VdsoDataPtr::V5_0(vdso_data_v5_0::vdsodata_ptr(vdso_addr)),
             (5, 3..=5) => VdsoDataPtr::V5_3(vdso_data_v5_3::vdsodata_ptr(vdso_addr)),
             (5, 6..=8) => VdsoDataPtr::V5_6(vdso_data_v5_6::vdsodata_ptr(vdso_addr)),
-            (5, _) => VdsoDataPtr::V5_9(vdso_data_v5_9::vdsodata_ptr(vdso_addr)),
-            (_, _) => return Err(()),
-        };
-
-        // If linux support a clockid, then we think vdso support it too temporaryly.
-        // We will verify whether vdso can support this clockid correctly later.
-        for &clockid in &clockids {
-            if !(tss[clockid as usize].tv_sec == 0 && tss[clockid as usize].tv_nsec == 0) {
-                support_clocks[clockid as usize] = true;
-            }
-        }
+            (5, 9..=12) => VdsoDataPtr::V5_9(vdso_data_v5_9::vdsodata_ptr(vdso_addr)),
+            (_, _) => return_errno!(EINVAL, "Vdso match kernel release failed"),
+        })
+    }
 
-        let mut vdso = Self {
-            vdso_data_ptr,
-            hres_resolution,
-            coarse_resolution,
-            support_clocks,
-        };
-
-        // Compare the results of Linux and vdso
-        // to check whether vdso can support the clockid correctly.
-        for &clockid in &clockids {
-            if vdso.support_clocks[clockid as usize] {
-                let mut tp = timespec {
-                    tv_sec: 0,
-                    tv_nsec: 0,
+    /// Compare the results of Linux syscall and vdso to check whether vdso can support the clockid correctly.
+    fn get_checked_vdso(vdso: Self) -> Result<Self> {
```

Comment:
Usually, a getter method is simple and do not change the state of an object. This method is complex and consumes `self`. So a `get_xxx` name is not a good fit for this method. How about `validate_accuracy` or `check_accuracy`?

Reply:
changed to `validate_accuracy`